File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/interop/testserver_2/vb/TestServer.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports System
Imports System.Reflection
Imports System.Runtime.CompilerServices

<assembly:AssemblyKeyFile("TestKey.snk")>
<assembly:AssemblyVersion("1.0.0.0")>

Namespace TestServer

	Public Interface ITest
		Function GetArray() As String()
	End Interface

	Public Class Test : Implements ITest
		Function GetArray() As String() Implements ITest.GetArray
			Dim stringArray() As String = {"one", "two", "three"}
			Return stringArray
		End Function
	End Class
	
End Namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/interop/testserver_2/cs/TestServer.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System;
using System.Reflection;
using System.Runtime.CompilerServices;

[assembly:AssemblyKeyFile("TestKey.snk")]
[assembly:AssemblyVersion("1.0.0.0")]

namespace TestServer
{
	public interface ITest
	{
		string[] GetArray();
	}		
	
	public class Test : ITest
	{
		string[] ITest.GetArray()
		{
			string[] stringArray = {"red", "yellow", "blue"};
			return stringArray;
		}		
	}
}
